core/kotlinx-coroutines-core/src/internal/Exceptions.kt

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright 2016-2018 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package kotlinx.coroutines.internal
+
+import kotlinx.coroutines.*
+import java.util.*
+import kotlin.coroutines.*
+import kotlin.coroutines.intrinsics.*
+
+internal actual fun <E : Throwable> recoverStackTrace(exception: E): E {
+    if (recoveryDisabled(exception)) {
+        return exception
+    }
+
+    val copy = tryCopyException(exception) ?: return exception
+    return copy.sanitizeStackTrace()
+}
+
+private fun <E : Throwable> E.sanitizeStackTrace(): E {
+    val size = stackTrace.size
+
+    var lastIntrinsic = -1
+    for (i in 0 until size) {
```

Comment:
Use `indexOfFirst { ... }` instead of the loop to clearly show what you are doing here and avoid `var`.